### ISSUE TITLE
Unable to find a way to customize the spinner and clearButton in asyncTypeHead.

### ISSUE THREAD

**Issue by Bharateshwar** [ericgio/react-bootstrap-typeahead#557]
Spinner refers to the spinner that appears on the input on the right corner when isLoading = {true}.
clearButton refers to the clearButton to clear data in input box.

**Comment by ericgio** [comment]
Hey @Bharateshwar, thanks for the question. You can customize those items by passing your own components via the typeahead's child render prop:

```jsx
  <Typeahead ... >
    {({ onClear, selected }) => (
      <div className="rbt-aux">
        {!!selected.length && <MyClearButton onClick={onClear} />}
        {!selected.length && isLoading && <MySpinner />}
      </div>
    )}
  </Typeahead>
```
Working example: https://codesandbox.io/s/rbt-custom-aux-components-gn3kn

Note that the `.rbt-aux` classname is used internally to position the built-in clear button and loader components and can be re-used. You may also add your own styles to position the custom components however you want.

**Comment by Bharateshwar** [comment]
hi @ericgio , I had a look at your solution and tried a few things ... but it's still not working. 
With the method you provided. 
It renders two spinners in the asynctypehead. 
Please help.

**Comment by ericgio** [comment]
Yeah with AsyncTypeahead the `isLoading` prop is automatically passed to the underlying Typeahead component, which triggers the built-in spinner.

One simple workaround would be to hide the built-in spinner using CSS. Alternatively, the latest v5 package includes a `useAsync` hook that would allow you to pass `isLoading` directly to your spinner rather than through the Typeahead component.

**Comment by Bharateshwar** [comment]
could you please give a working example.

**Comment by kimfucious** [comment]
I know this is closed, but I'm sharing a solution, as it may help others.

I'm using SCSS, so in my `custom.scss` file, I did the following:
```scss
.my-rbt-aux {
  @extend .rbt-aux;
  right: 75px;
  width: auto;
}
.rbt-loader {
  display: none;
}
```
☝️ Hiding `.rbt-loader` is the droid I believe @Bharateshwar is looking for.

In the component where I use `AsyncTypeahead` I do the below, where `<Spinner />` is a custom spinner I made using [React Spinners](https://www.npmjs.com/package/react-spinners). 
```js
  <AsyncTypeahead
    aria-describedBy="thaiAddressHelper"
    id="thai-postal-code-search-input"
    isLoading={isSpinning}
    labelKey={(option) =>
      `${option.district_en}, ${option.amphoe_en}, ${option.province_en}  ${option.postalCode}`
    }
    minLength={5}
    inputProps={{ maxLength: 5 }}
    onChange={handleChange}
    onSearch={handleSearch}
    options={options}
    placeholder="e.g. 50200"
    ref={searchRef}
    selected={selectedAddress}
  />
  {isSpinning ? (
    <div className="my-rbt-aux">
      <Spinner size={10} color="primary" type="beat" />
    </div>
  ) : null}
  <div className="input-group-append">
    <button
      className="btn btn-outline-primary"
      onClick={() => {
        setSelectedAddress([]);
        setOptions([]);
        searchRef.current.clear();
      }}
    >
      Clear
    </button>
  </div>
```
To me this seems a bit inelegant as I'm futzing with the absolute placement to make the spinner appear to the left of the `input-group-append`, but I can't figure out a better way to do it yet.  If I leave the absolute placement as is, the spinner shows up on top of the clear button.  I'm pretty sure, I'm not doing that right 😉 

I'm not sure if @ericgio reads closed issues, so any advice from anyone would be appreciated.

**Comment by ericgio** [comment]
@kimfucious, please see my initial answer on this thread regarding passing the spinner in as a child. That should prevent you from needing to add additional positioning. Ie:

```jsx
<AsyncTypeahead ... >
  {isSpinning ? (
    <div className="my-rbt-aux">
      <Spinner size={10} color="primary" type="beat" />
    </div>
  ) : null}
</AsyncTypeahead>
```

**Comment by kimfucious** [comment]
Hi @ericgio,

Thanks for the kind follow up.

My issue with positioning, I learned in hindsight, was that I was using an unnecessary `input-group-append`.

In brief, I did not realize that there was a `clearButton` that I could add as a prop.

Having learned that, I got rid of that unnecessary button, and worked with what the lib delivers well. 

I still needed to nudge the spinner over to the left a bit with:
```scss
.my-rbt-aux {
  @extend .rbt-aux;
  right: 10px;
  width: auto;
}
```
But that's expected, and the end solution is nice.

Again, excellent work with this awesome library!
